bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/XMLText.java

```diff
@@ -57,6 +58,21 @@ public String getItemType() {
         return getNodeType().value();
     }
 
+    @Override
```

Comment:
Shouldn't we override `attemptFreeze()`, `freezeDirect()` and `freeze()` too?

Reply:
We can re-use the super class methods without any ill efect IMO

Reply:
Aren't they unnecessary though?

Reply:
They are sitting high in the hierarchy, providing expected behavior.
In my view, overriding them with empty methods are unnecessary.

Reply:
`freezeDirect()` is OK. 

But for `attemptFreeze()` if (by the slightest chance) there is an attempt to freeze the same value concurrently there is a chance it'll panic. https://github.com/ballerina-platform/ballerina-lang/blob/master/bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/freeze/FreezeUtils.java#L45 unless we set the frozen flag when creating the value itself.

This IMO is unnecessary since it's making it read-only is always possible.

And for `freeze()` I think we can remove `freezeDirect()` call.

```
    @Override
    public Object freeze() {
        // freezeDirect(); we can remove this
        return this;
    }
```

Reply:
If I'm not mistaken freeze action could only be initiated from a `cloneReadOnly` operation, which means there couldn't be any concurrent invocations

Reply:
I guess it'll work since we seem to be checking `isFrozen()` on ` copy()` results before attempting freezing and  `.cloneReadOnly()` seems to be calling `freezeDirect()`.